eclair-core/src/test/scala/fr/acinq/eclair/wire/protocol/OfferTypesSpec.scala

```diff
@@ -157,80 +115,60 @@ class OfferTypesSpec extends AnyFunSuite {
       assert(!noChain.isValidFor(offer))
-      val otherChain = signInvoiceRequest(request1.copy(records = TlvStream(request1.records.records.map { case Chain(_) => Chain(Block.LivenetGenesisBlock.hash) case x => x })), payerKey)
+      val otherChain = signInvoiceRequest(request1.copy(records = TlvStream(request1.records.records.map { case InvoiceRequestChain(_) => InvoiceRequestChain(Block.LivenetGenesisBlock.hash) case x => x })), payerKey)
       assert(!otherChain.isValidFor(offer))
     }
   }
 
   test("check that invoice request matches offer (multiple items)") {
     val offer = Offer(TlvStream(
-      Amount(500 msat),
-      Description("offer for multiple items"),
-      NodeId(randomKey().publicKey),
-      QuantityMin(3),
-      QuantityMax(10),
+      OfferAmount(500 msat),
+      OfferDescription("offer for multiple items"),
+      OfferNodeId(randomKey().publicKey),
+      OfferQuantityMax(10),
     ))
     val payerKey = randomKey()
     val request = InvoiceRequest(offer, 1600 msat, 3, Features.empty, payerKey, Block.LivenetGenesisBlock.hash)
-    assert(request.records.get[Quantity].nonEmpty)
+    assert(request.records.get[InvoiceRequestQuantity].nonEmpty)
     assert(request.isValidFor(offer))
     val invalidAmount = InvoiceRequest(offer, 2400 msat, 5, Features.empty, payerKey, Block.LivenetGenesisBlock.hash)
     assert(!invalidAmount.isValidFor(offer))
-    val tooFewItems = InvoiceRequest(offer, 1000 msat, 2, Features.empty, payerKey, Block.LivenetGenesisBlock.hash)
-    assert(!tooFewItems.isValidFor(offer))
     val tooManyItems = InvoiceRequest(offer, 5500 msat, 11, Features.empty, payerKey, Block.LivenetGenesisBlock.hash)
     assert(!tooManyItems.isValidFor(offer))
   }
 
-  test("decode invoice request") {
-    val encoded = "lnr1qvsxlc5vp2m0rvmjcxn2y34wv0m5lyc7sdj7zksgn35dvxgqqqqqqqqyypz8xu3xwsqpar9dd26lgrrvc7s63ljt0pgh6ag2utv5udez7n2mjzqzz47qcqczqgqzqqgzycsv2tmjgzc5l546aldq699wj9pdusvfred97l352p4aa862vqvzw5p8pdyjqctdyppxzardv9hrypx74klwluzqd0rqgeew2uhuagttuv6aqwklvm0xmlg52lfnagzw8ygt0wrtnv2tsx69m6tgug7njaw5ypa5fn369n9yzc87v02rqccj9h04dxf3nzc"
-    val Success(request) = InvoiceRequest.decode(encoded)
-    assert(request.amount == Some(5500 msat))
-    assert(request.offerId == ByteVector32(hex"4473722674001e8cad6ab5f40c6cc7a1a8fe4b78517d750ae2d94e3722f4d5b9"))
-    assert(request.quantity == 2)
-    assert(request.features == Features(VariableLengthOnion -> Optional, BasicMultiPartPayment -> Optional))
-    assert(request.records.get[Chain].nonEmpty)
-    assert(request.chain == Block.LivenetGenesisBlock.hash)
-    assert(request.payerKey == ByteVector32(hex"c52f7240b14fd2baefda0d14ae9142de41891e5a5f7e34506bde9f4a60182750"))
-    assert(request.payerInfo == Some(hex"deadbeef"))
-    assert(request.payerNote == Some("I am Batman"))
-    assert(request.encode() == encoded)
-  }
-
   test("decode invalid invoice request") {
     val testCases = Seq(
       // Missing offer id.
```

Comment:
This test doesn't make sense anymore since offer id doesn't exist? Generally it looks like this test suite requires a careful rework to remove obsolete stuff and make sure we're correctly testing against the latest specification.

Reply:
Removed obsolete tests.